File: javascript/leap/leap.js
var Year = function(year) {
  this.year = year;
};

Year.prototype.isLeap = function() {
  if (isNaN(this.year)) {
    return false;
  }

  if (this.year % 4 === 0) {
    if (this.year % 400 === 0) {
      return true;
    }

    if (this.year % 100 === 0) {
      return false;
    }

    return true;
  }

  return false;
};

module.exports = Year;

File: python/leap/leap.py
def is_leap_year(year):
    if not isinstance(year, (int, long)):
        return False
    else:
        if year % 4 == 0:
            if year % 400 == 0:
                return True
            if year % 100 == 0:
                return False
            return True
